src/models/input_ids.h

```diff
@@ -4,19 +4,26 @@
 
 namespace Generators {
 
-struct InputIDs {
+struct InputIDsInterface {
```

Comment:
Can we have this base class just be 'InputIDs' and then we'd have two types of them, the non sliding window and the InputIDsSlidingWindow. How do we refer to the non sliding window ones?